examples/zephyr/certificate_provisioning/pytest/test_sample.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import pytest
+import subprocess
+import datetime
+import logging
+from time import sleep
+
+LOGGER = logging.getLogger(__name__)
+PROJ_NAME = 'firmware_ci'
+
+FS_SUBDIR = "/lfs1/credentials"
+FS_CRT_PATH = f"{FS_SUBDIR}/client_cert.der"
+FS_KEY_PATH = f"{FS_SUBDIR}/private_key.der"
+
+pytestmark = pytest.mark.anyio
+
+def subprocess_logger(result, log_msg):
+    if result.stdout:
+        LOGGER.info(f'{log_msg} stdout: {result.stdout}')
+    if result.stderr:
+        LOGGER.error(f'{log_msg} stderr: {result.stderr}')
+
+async def test_credentials(shell, project, device_name, device_port, certificate_cred):
+
+    # Ensure there are no credentials currently stored on device
+
+    shell._device.readlines_until(regex=".*Start certificate provisioning sample", timeout=90.0)
+    sleep(5) # Time for fs bringup and in case nRF9160 reset loop reboot happens
+    shell.exec_command(f'fs rm {FS_CRT_PATH}')
+    shell.exec_command(f'fs rm {FS_KEY_PATH}')
+    shell.exec_command(f'fs rm {FS_SUBDIR}')
+    shell._device.clear_buffer()
+    shell._device.write('kernel reboot cold\n\n'.encode())
+    shell._device.readlines_until(regex=".*Start certificate provisioning sample", timeout=90.0)
+    sleep(5) # Time for fs bringup and in case nRF9160 reset loop reboot happens
+
+    # Check cloud to verify device does not exist
+
+    with pytest.raises(Exception):
+        device = await project.device_by_name(device_name)
+
+    # Generate device certificates
+
+    root_crt_id = certificate_cred  # Cause harness to generate root certs
```

Comment:
NIT: technically the harness generates the root cert before this function, not when we assign the variable here.